pkg/operators/process/process.go

```diff
@@ -0,0 +1,778 @@
+				if err1 == nil && err2 == nil {
+					totalTime := utime + stime
+
+					// Calculate CPU usage percentage
+					if prevTime, ok := p.lastCPUTimes[pid]; ok && timeDelta > 0 {
+						// CPU usage is the difference in CPU time (in jiffies) divided by the elapsed time (in seconds)
+						// We need to convert jiffies to seconds by dividing by clockTicksPerSecond
+						cpuTimeDiff := float64(totalTime - prevTime)
+						if cpuTimeDiff < 0 {
+							// Handle case where process was restarted with same PID
+							cpuTimeDiff = float64(totalTime)
+						}
+						cpuUsage := cpuTimeDiff / clockTicksPerSecond / timeDelta * 100.0
+						procInfo.CPUUsage = cpuUsage
+					} else {
+						procInfo.CPUUsage = 0.0
+					}
+
+					// Store current CPU time for next calculation
+					p.lastCPUTimes[pid] = totalTime
+				}
+			}
+
+			// Get start time if needed
+			if p.enabledFields[fieldStartTime] {
+				// Field 22 is the start time (in clock ticks since system boot)
+				if startTime, err := strconv.ParseUint(statFields[21], 10, 64); err == nil {
+					procInfo.StartTime = startTime
+
+					// Convert clock ticks to seconds since boot
+					startTimeSec := float64(startTime) / clockTicksPerSecond
+
+					// Calculate the actual start time by adding to boot time
+					procInfo.StartTimeStr = p.bootTime.Add(time.Duration(startTimeSec * float64(time.Second)))
+				}
+			}
+		}
+	}
+
+	return procInfo, nil
+}
+
+// unescapeCommand unescapes the command string according to kernel escaping rules
+// %ESCAPE_SPACE: ('\f', '\n', '\r', '\t', '\v')
+// %ESCAPE_SPECIAL: ('\"', '\\', '\a', '\e')
+func unescapeCommand(cmd string) string {
+	// If there are no escape sequences, return the original string
+	if !strings.Contains(cmd, "\\") {
+		return cmd
+	}
+
+	var builder strings.Builder
+	builder.Grow(len(cmd)) // Pre-allocate capacity to avoid reallocations
+
+	for i := 0; i < len(cmd); i++ {
+		// Check for escape sequence
+		if cmd[i] == '\\' && i+1 < len(cmd) {
+			switch cmd[i+1] {
+			// Special whitespace characters
+			case 'f':
+				builder.WriteByte('\f')
+			case 'n':
+				builder.WriteByte('\n')
+			case 'r':
+				builder.WriteByte('\r')
+			case 't':
+				builder.WriteByte('\t')
+			case 'v':
+				builder.WriteByte('\v')
+			// Special characters
+			case '"':
+				builder.WriteByte('"')
+			case '\\':
+				builder.WriteByte('\\')
+			case 'a':
+				builder.WriteByte('\a')
+			case 'e':
+				builder.WriteByte(27) // ASCII 27 for escape character
+			default:
+				// Unknown escape sequence, keep both characters
+				builder.WriteByte('\\')
+				builder.WriteByte(cmd[i+1])
+			}
+			i++ // Skip the next character as we've already processed it
+		} else {
+			builder.WriteByte(cmd[i])
+		}
+	}
+
+	return builder.String()
+}
+
+// parseStatFile parses the /proc/[pid]/stat file, handling the comm field correctly
+// The comm field is enclosed in parentheses and may contain spaces
+func parseStatFile(statData string) ([]string, error) {
```

Comment:
Since `os.ReadFile` gives a `[]byte`,this function could work on a `[]byte` to avoid memory copies:
```suggestion
func parseStatFile(statData []byte) ([]string, error) {
```
The [bytes package](https://pkg.go.dev/bytes) has similar functions for Index, LastIndex, TrimLeft, etc.